releasenotes.txt

```diff
@@ -1,4 +1,155 @@
+   * #1535 Fix some HQL functions registration
+   * #1534 Fixed entity name retrieval for EntityProjection
+   * #1526 ExpressionKeyVisitor does not produce unique keys for anonymous types coming from different assemblies
+   * #1514 Fix exceptions serialization
+   * #1511 Test Unicode string.
+   * #1509 Add missing NHibernateLogLevel.Info in example web project
+   * #1507 NH-3119 - fix test not supporting optimization
+   * #1506 SQLite is bugged with distributed transactions: disable distributed t…
+   * #1505 Chaining scopes with ODBC is bugged: disabling the test.
+   * #1501 Fix NH-3023 test
+   * #1496 Fix ManyToOneType.IsModified to handle both object instance and identifier passed to the parameter “old”.
+   * #1491 Forgotten async generation for #1487
+   * #1486 Fix IsModified so that a null equates empty components when using select-before-update.
+   * #1484 Fix default types
+   * #1478 Exception when using envers with the latest logging changes
+   * #1468 Comparison with DateTime? produces wrong SQL
+   * #1463 Fix a null reference case in session context
+   * #1454 Fix ProxyFactory cache
+   * #1445 Upgrade AsyncGenerator to 0.6.2 and regenerate.
+   * #1442 Unable to use an entity with a `FieldInterceptor` property and a lazy loaded property
+   * #1436  StackOverflowException  in Merge  Entity With Lazy Property   
+   * #1434 Replace remaining SetOptions with WithOptions
+   * #1372 NH-3982 - Simple query with Cacheable, Fetch and SingleOrDefault throws exception (regression from 3.3.0)
+   * #1371 NH-3898 - Configuring a property with generated="insert" turns "Property.IsUpdatable" into"false" even using update="true" in the xml mapping file.
+   * #1363 NH-2500 - NH 3.0 Linq provider uses query parameters from first call in subsequent calls.
+   * #1335 NH-3787 - Decimal truncation in Linq ternary expression
+   * #1330 NH-3673 - Closure variable values locked in from expressions in NHibernate LINQ provider
+   * #1226 NH-2534 - Join-fetching a many-to-one with property-ref results in select n+1 problem
+   * #1196 NH-4087 - Decimal truncation occurs after 5 digits
+   * #1119 NH-3084 - Class NHibernate.Loader.Loader logs SQL statement on INFO level
+   * #1052 NH-3976 - Inconsistent Decimal/NHibernateUtil.Currency handling causing runtime error when using Oracle.ManagedDataAccess
+   * #987 NH-1509 - MsSql2000Dialect does not use default schema when creating "if exists" statement
+   * #448 NH-1285 - Drop/Create script with default_schema/default_catalog fix(SqlServer)
+
+** New Feature
+
+   * #1588 Add a generic batcher for insert/update/delete statements, usable with PostgreSQL and others
+   * #1545 Support to join not associated entities in Criteria (aka Entity Join)
+   * #1451 New StaticProxyFactoryFactory
+   * #1403 Add timeouts support to MultiCriteria
+   * #1377 Logging refactoring
+   * #954 NH-3807 - Support for .NET Core 2.0
+   * #948 NH-3435 - Ability to select the root entity in a criteria projection
```

Comment:
I think this issue needs to be renamed to "Ability to select entities in a criteria projection" . As implementation #1471 is not limited to root entity.

Reply:
Done.